backend/shared/decorators.py

```diff
@@ -0,0 +1,56 @@
+import os
+from flask import request, jsonify
+from functools import wraps
+from azure.keyvault.secrets import SecretClient
+from azure.identity import DefaultAzureCredential
+
+
+def get_secret(secretName):
```

Comment:
@PedroLabrador This function is being used in multiple places in the code. We might want to consider defining it in a single location.

take a look at the definition here:
backend/financial_doc_processor.py